src/poligrain/spatial.py

```diff
@@ -86,17 +86,91 @@ def project_point_coordinates(
     return x_projected, y_projected
 
 
+def get_closest_points_to_point(
+    ds_points: xr.DataArray | xr.Dataset,
+    ds_points_neighbors: xr.DataArray | xr.Dataset,
+    max_distance: float,
+    n_closest: int,
+) -> xr.Dataset:
+    """Get the closest points for given point locations.
+
+    Note that both datasets that are passed as input have to have the
+    variables `x` and `y` which should be projected coordinates that preserve
+    lengths as good as possible.
+
+    Parameters
+    ----------
+    ds_points : xr.DataArray | xr.Dataset
+        This is the dataset for which the nearest neighbors will be looked up. That is,
+        for each point location in this dataset the nearest neighbors from
+        `ds_points_neighbors` will be returned.
+    ds_points_neighbors : xr.DataArray | xr.Dataset
+        This is the dataset from which the nearest neighbors will be looked up.
+    max_distance : float
+        The allowed distance of neighbors has to be smaller than `max_distance`.
+        The unites are the units used for the projected coordinates `x` and
+        `y` in the two datasets.
+    n_closest : int
+        The maximum number of nearest neighbors to be returned.
+
+    Returns
+    -------
+    xr.Dataset
+        A dataset which has `distance` and `neighbor_id` as variables along the
+        dimensions `id`, taken from `ds_points` and `n_closest`. The unit of the
+        distance follows from the unit of the projected coordinates of the input
+        datasets. The `neighbor_id` entries for point locations that are further
+        away then `max_distance` are set to NaN.
+    """
+    x, y = get_point_xy(ds_points)
+    x_neighbors, y_neighbors = get_point_xy(ds_points_neighbors)
+    tree_neighbors = scipy.spatial.KDTree(
+        data=list(zip(x_neighbors.values, y_neighbors.values))
+    )
+    distances, ixs = tree_neighbors.query(
+        list(zip(x.values, y.values)),
+        k=n_closest,
+        distance_upper_bound=max_distance,
+    )
```

Comment:
when passing n_closest=1 to the function this line retruns a 1D array instead of the required 2D array, causing an error

Reply:
I solved this for the 1D case (where passing n_closest = 1 returned an integer using np.atleast_1d(ind_nearest_gauges). there should be an 2D equivalent, or you will need to do a test and then reshape(-1, 1)

Reply:
thanks. good catch. I will add a test and then solve probably using your solution provided in the comment below.